tests/core/contracts/test_contract_call_interface.py

```diff
@@ -2313,3 +2325,12 @@ async def test_async_changing_default_block_identifier(async_w3, async_math_cont
     assert (
         await async_math_contract.functions.counter().call(block_identifier=None) == 7
     )
+
+
+def test_async_functions_iterator(async_w3, async_math_contract):
```

Comment:
No async stuff happening in here, is it worth having this separate test?

Reply:
It has the Async type now. Resolving.